src/borg/testsuite/repo_new.py

```diff
@@ -1,34 +1,51 @@
 import os
 import shutil
 from contextlib import contextmanager
+from unittest.mock import patch
 
 import pytest
+from borg.hashindex import NSIndex
 
-from ..helpers import IntegrityError
-from ..repository import Repository, MAX_DATA_SIZE, TAG_DELETE, TAG_PUT2, TAG_PUT, TAG_COMMIT, MAGIC
+from ..helpers import IntegrityError, msgpack
+from ..locking import Lock, LockFailed
+from ..repository import Repository, MAX_DATA_SIZE, TAG_DELETE, TAG_PUT2, TAG_PUT, TAG_COMMIT, MAGIC, LoggedIO
 from ..repoobj import RepoObj
 from .hashindex import H
 
 UNSPECIFIED = object()  # for default values where we can't use None
-EXCLUSIVE = True
+exclusive = UNSPECIFIED
+
+
+def get_exclusive():
+    global exclusive
+    if exclusive is True:
+        return True
+    elif exclusive is False:
+        return False
+    else:
+        return UNSPECIFIED
+
+
+def set_exclusive(value):
+    global exclusive
+    exclusive = value
```

Comment:
These `get` and `set` are kind of a hacky way of changing the `exclusive` value, which some tests need changed when reopening a repository. I am working to find a more elegant way of doing this, but for now it works.